jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/BulkLoaderX.java

```diff
@@ -41,6 +56,26 @@
      */
     public static boolean CompressSortIndexFiles = true;
 
+    // Ubuntu: it now (21.04) is at /usr/bin/gzip.
+    //   /bin has become a symbolic link to /usr/bin.
+    //   New installs of 20.04 have it at /usr/bin, upgrades have it at /bin.
+    /*package*/ static String gzipProgram() {
+        if ( programInstalledAt("/usr/bin/gzip") )
```

Comment:
Since `programInstalledAt` throws an exception if the given path is not executable won't this never try the other option in that case?

Reply:
Yes, but if gzip exists at /usr/bin/gzip and at /bin/gzip and only one is executable, something weird is going on. The order of lookup is "/usr/bin" then "/bin" for Ubuntu 20.xx upgrades. The execution is for a "shouldn't happen" situation caught early rather than when sort runs.